lib/internal/util/inspect.js

```diff
@@ -1320,12 +1323,50 @@ function markCwd(ctx, line, workingDirectory) {
   return tempLine;
 }
 
-function safeGetCWD() {
+// The following characters are percent-encoded when converting from file path
+// to URL:
+// - %: The percent character is the only character not encoded by the
+//        `pathname` setter.
+// - \: Backslash is encoded on non-windows platforms since it's a valid
+//      character but the `pathname` setters replaces it by a forward slash.
+// - LF: The newline character is stripped out by the `pathname` setter.
+//       (See whatwg/url#419)
+// - CR: The carriage return character is also stripped out by the `pathname`
+//       setter.
+// - TAB: The tab character is also stripped out by the `pathname` setter.
+const percentRegEx = /%/g;
+const backslashRegEx = /\\/g;
+const newlineRegEx = /\n/g;
+const carriageReturnRegEx = /\r/g;
+const tabRegEx = /\t/g;
+
+function encodePathChars(filepath) {
+  if (StringPrototypeIncludes(filepath, '%'))
+    filepath = StringPrototypeReplace(filepath, percentRegEx, '%25');
+  // In posix, backslash is a valid character in paths:
+  if (!isWindows && StringPrototypeIncludes(filepath, '\\'))
+    filepath = StringPrototypeReplace(filepath, backslashRegEx, '%5C');
+  if (StringPrototypeIncludes(filepath, '\n'))
+    filepath = StringPrototypeReplace(filepath, newlineRegEx, '%0A');
+  if (StringPrototypeIncludes(filepath, '\r'))
+    filepath = StringPrototypeReplace(filepath, carriageReturnRegEx, '%0D');
+  if (StringPrototypeIncludes(filepath, '\t'))
+    filepath = StringPrototypeReplace(filepath, tabRegEx, '%09');
+  return filepath;
+}
+
+function safeGetEscapedCWD(stackFramePart) {
+  let workingDirectory;
   try {
-    return process.cwd();
+    workingDirectory = process.cwd();
   } catch {
-    // Explicitly empty
+    return;
+  }
+  // Handle ESM
+  if (stackFramePart.includes('file://')) {
```

Comment:
```suggestion
  if (StringPrototypeIncludes(stackFramePart, 'file://')) {
```

Reply:
There can be false positive with that check, should we worry about this? I suppose it's very unlikely to happen in the real world, but we should make sure that the error stack trace is still readable in the edge case.

Reply:
I understand this is a best effort and it's definitely OK not to get right. The issue is that if there's a mix use of CJS and ESM in the stack trace (or if there's a function named e.g. `file://` in the trace) , now the CJS paths are not being greyed out at all:

```sh
mkdir repro
cd repro
echo 'module.exports = () => { throw new Error }' > file.cjs
echo 'import t from "./file.cjs";t()' > entry.mjs
../node entry.mjs
cd ..
rm -r repro
```

Maybe a more robust alternative would be to always compute both formats (path and file URL), and attempt to grey out both of them. Note that I don't consider this blocking, it's fine to land as is if what I'm suggest is more complicated than I think.